### PR TITLE
[17.09] Add augustus and snap datatypes

### PR DESCRIPTION
**Pull request by abretaud**
Hey,
As discussed in https://github.com/galaxyproject/tools-iuc/pull/1503 2 new datatypes for Augustus and SNAP prediction models.
Maybe I'm a bit optimistic targeting 17.09, tell me if I should target dev branch instead

### CODE REVIEW

File: config/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -679,6 +682,7 @@
     <sniffer type="galaxy.datatypes.binary:Fast5ArchiveGz" />
     <sniffer type="galaxy.datatypes.binary:Fast5ArchiveBz2" />
     <sniffer type="galaxy.datatypes.binary:Fast5Archive" />
+    <sniffer type="galaxy.datatypes.annotation:Augustus" />
```
Comment (bgruening) [inline config/datatypes_conf.xml.sample:685]: should snaphmm be added to the sniffer as well?

**Comment by abretaud** [comment]
Ah yes, thanks!
I'd need the tools for a training next week, do you think there is some chance to have it merged before, or not? (I can always make a dirty local install if needed)

File: lib/galaxy/datatypes/annotation.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import tarfile
+
+from galaxy.datatypes.binary import Binary, CompressedArchive
+from galaxy.datatypes.data import get_file_peek, Text
+from galaxy.util import nice_size
+
+log = logging.getLogger(__name__)
+
+
+class SnapHmm(Text):
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/annotation.py:11]: Any `edam_format` or `edam_data` that could be added (also for Augustus)?

Comment (abretaud) [reply lib/galaxy/datatypes/annotation.py:11]: For snap, I could add http://edamontology.org/data_1364, but for augustus I didn't find anything more specific than http://edamontology.org/data_0950, do you think it's worth adding it anyway?

Comment (nsoranzo) [reply lib/galaxy/datatypes/annotation.py:11]: Not an expert, but I suppose it's better than nothing.

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**